### PR TITLE
✨ feat: Prehistory of Ethereum

### LINKED ISSUE #152: Wiki - Protocol Overview
The Protocol overview page should give an introduction to Ethereum from technical perspective and help reader navigate through the wiki. 

https://epf.wiki/#/wiki/protocol/overview

I am not settled on what the content should exactly look like, I would be glad for inputs and collaborators on this. Kind of an updated, simplified whitepaper describing the protocol comes to mind. Generally it can be guided by the structure of the protocol with a short overview of various domains.

### PR DESCRIPTION
**Pull request by raxhvl**
## Wiki PR Checklist

closes #152 

- [x] Background on peer to peer computing.
- [x] Introduction on bitcoin.
- [x] Early bitcoin apps and their limitation.
- [x] Introduction of Ethereum and its world computer.
  - [x] How Ethereum world computer compares to traditional computers.
- [x] Applications

 There are some typos in `docs/wiki/CL/cl-clients.md`. Let me know if you want me to fix them.

### CODE REVIEW

File: docs/wiki/protocol/overview.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Ethereum Overview
```
Comment (taxmeifyoucan) [inline docs/wiki/protocol/overview.md:1]: I would change this doc to Intro to Ethereum and have another doc for Protocol Overview (high level explanation, EL/CL, usage, etc)

File: docs/wiki/protocol/overview.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+## Network coming online and peer-to-peer computing
+
+The concept of peer-to-peer (P2P) networks, where computers communicate and share resources directly without a central authority, has a rich history dating back to the 1960s. Pioneering projects like the ARPANET project, switched on in late October 1969 became the first wide-area packet-switched network, laid the groundwork for this decentralized approach. ARPANET demonstrated the model of distributed communication, where "every participating node could request and serve content".
```
Comment (taxmeifyoucan) [inline docs/wiki/protocol/overview.md:5]:
```suggestion
The concept of peer-to-peer (P2P) networks, where computers communicate and share resources directly without a central authority, has a rich history dating back to the 1960s. Pioneering projects like the ARPANET, switched on in late October 1969 became the first wide-area packet-switched network, laid the groundwork for this decentralized approach. ARPANET demonstrated the model of distributed communication, where "every participating node could request and serve content".
```

File: docs/wiki/protocol/overview.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+![ARPANET-map](img/overview/ARPANET-map.jpg)
+> The map of ARPANET in the 1960s
+
+This decentralized philosophy gained significant traction in the late 1990s with the rise of file-sharing platforms like [Napster](https://en.wikipedia.org/wiki/Napster) (1999). These platforms challenged the traditional client-server model, where users relied on a central server to access data. Napster, and subsequent platforms like [BitTorrent](https://en.wikipedia.org/wiki/BitTorrent) (2001), sparked new conversations and philosophies highlighting the potential for a more democratic and distributed online interaction.
```
Comment (taxmeifyoucan) [inline docs/wiki/protocol/overview.md:10]: It's bit of jump from arpanet to torrents :) To bridge this a little:

I would emphasize that internet predecessor, arpanet, was basically a decentralized network because of its simple structure. Later as internet evolves, and with the WWW, the client-server model becomes dominant. P2P networks are gaining traction for distributed technologies..

File: docs/wiki/protocol/overview.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## Introduction of Bitcoin
+
+Introduced in a 2008 paper by Satoshi Nakamoto titled ["Bitcoin: A Peer-to-Peer Electronic Cash System"](https://bitcoin.org/bitcoin.pdf), blockchain established a distributed ledger system where data is cryptographically linked in chronological blocks. This innovation empowered the creation of Bitcoin, the first decentralized digital currency, eliminating the need for trusted third-party intermediaries like banks.
```
Comment (taxmeifyoucan) [inline docs/wiki/protocol/overview.md:14]: Bitcoin is built on more concepts than just p2p tech, I would mention add a section or add to previous one something on cryptography, initial attempts for digital cash by cypherpunks..

File: docs/wiki/protocol/overview.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+Bitcoin's focus on secure digital payments, while revolutionary, limited its ability to run more complex applications. Similar to how modern applications rely on HTTP for communication, early projects attempted to expand Bitcoin's functionality by leveraging its secure cryptographic and peer-to-peer foundation.
```
Comment (taxmeifyoucan) [inline docs/wiki/protocol/overview.md:16]: I don't really get the HTTP comparison here 😅

Bitcoin transactions are based on satisfying a simple script which doesn't allow full programmability, it is limited only to enable transfers. Projects inspired by this technology try to utilize it for different use cases..

File: docs/wiki/protocol/overview.md (hunk 1 above)
Comment (taxmeifyoucan) [reply docs/wiki/protocol/overview.md:1]: maybe Intro and history

File: docs/wiki/protocol/overview.md (hunk 5 above)
Comment (raxhvl) [reply docs/wiki/protocol/overview.md:16]: I was going for an analogy between how modern apps ( mobile, web) use HTTP as a buliding block, namecoin etc tried to use bitcoin as a building block. Does that make sense?

File: docs/wiki/protocol/overview.md (hunk 3 above)
Comment (raxhvl) [reply docs/wiki/protocol/overview.md:10]: Will add this. I was worried if too much history was being covered so I cut it short.

File: docs/wiki/protocol/overview.md (hunk 4 above)
Comment (raxhvl) [reply docs/wiki/protocol/overview.md:14]: Noted. That makes me think this page overlaps with history of Ethereum. Should we get rid of that page?

Comment (taxmeifyoucan) [reply docs/wiki/protocol/overview.md:14]: Yes, let's keep this as Prehistory+intro and the History page should track the evolution of the protocol itself

File: docs/wiki/protocol/overview.md (hunk 5 above)
Comment (taxmeifyoucan) [reply docs/wiki/protocol/overview.md:16]: Ah I see what you mean. I would rephrase it, something along what I said above, 'other projects utilize the fundamental technology of btc to apply to other use cases, similarly how different applications use the internet protocol...'

Comment (taxmeifyoucan) [reply docs/wiki/protocol/overview.md:16]: The important part here is that bitcoin scripting is limited and other uses of its blockchain proved impractical requiring drastic changes which are not compatible with the base protocol, therefore new separate blockchains like Eth spin out to it's own projects. Hope it makes sense :)

File: docs/wiki/protocol/overview.md (hunk 1 above)
Comment (Tomi-3-0) [reply docs/wiki/protocol/overview.md:1]: History and Introduction to Ethereum sounds good

**Comment by raxhvl** [comment]
I rewrote the entire article from non technical standpoint. 

Let me nerdsnipe this back to you @taxmeifyoucan. Grab coffee :coffee:

File: docs/wiki/protocol/prehistory.md [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+As of the time of writing, Ethereum has a market capitalization of **$400 billion.**
+
+> 📄 READ: [Vitalik's post about the origin of Ethereum.](https://vitalik.eth.limo/general/2017/09/14/prehistory.html)
+
+> 🎦 WATCH: [Mario Havel talks about the Ethereum philosophy.](https://streameth.org/ethereum_protocol_fellowship/watch?session=65d77e4f437a5c85775fef9d)
+
+> 📄 READ: [Evolution of Ethereum.](/wiki/protocol/history.md)
+
+## Resources
```
Comment (taxmeifyoucan) [inline docs/wiki/protocol/prehistory.md:292]: I would add recommendation for movie Revolution OS and maybe this https://www.youtube.com/watch?v=TdmpAy1hI8g talk by Tim May himself

**Review by taxmeifyoucan — APPROVED** [review]
this is awesome, thanks a lot Rahul! I had a blast reading it, really great overview of all the important pieces :)

**Comment by Tomi-3-0** [comment]
Great job! @raxhvl

File: docs/wiki/protocol/prehistory.md (hunk 6 above)
Comment (raxhvl) [reply docs/wiki/protocol/prehistory.md:301]: I have added both documentaries.

Plus this:

https://github.com/eth-protocol-fellows/protocol-studies/blob/a2fe6bd5b31754c6f2e7804db30a4930cc1776c4/docs/wiki/protocol/prehistory.md?plain=1#L47

**Comment by taxmeifyoucan** [comment]
image

**Comment by taxmeifyoucan** [comment]
image
